src/call/mod.rs

```diff
@@ -330,9 +380,10 @@ impl BatchContext {
 
     /// Get the status of the rpc call.
     pub fn rpc_status(&self) -> RpcStatus {
-        let status =
-            unsafe { grpc_sys::grpcwrap_batch_context_recv_status_on_client_status(self.ctx) };
-        let details = if status == RpcStatusCode::GRPC_STATUS_OK {
+        let status = RpcStatusCode::from(unsafe {
```

Comment:
Are we likely to get values outside the valid range? If not we should `assert` that status is valid, if they might be we might want to at least log invalid values